e3db/tests/test_share_integration.py

```diff
@@ -0,0 +1,111 @@
+import e3db
+import os
+import binascii
+import pytest
+import time
+import e3db.types
+import hashlib
+
+token = os.environ["REGISTRATION_TOKEN"]
+api_url = os.environ["DEFAULT_API_URL"]
+
+
+def crypto_mode():
+    if 'CRYPTO_SUITE' in os.environ and os.environ['CRYPTO_SUITE'] == 'NIST':
+        return 'nist'
+
+    return 'sodium'
+
+
+class TestShareIntegration():
+    @classmethod
+    def setup_class(self):
+        """
+        Setup where we create two clients using registration tokens to associate
+        them with out Innovault account. They will be used to test operations
+        later in the integration tests where we don't want to create a new client
+        every time.
+        """
+        client1_public_key, client1_private_key = e3db.Client.generate_keypair()
```

Comment:
Do the integration tests in general not have some clients already registered, or is this our standard set up style of all test classes?

Reply:
They do, I don't believe the unit tests are self contained though: e.g. if records are shared they are unshared after the tests ends. So I didn't want to have interaction between previous clients and these new tests.